### PR TITLE
feat: enable simde in package.xml

### PR DESCRIPTION
**Pull request by wep21**
I added libsimde-dev into package.xml.
Also I created [a PR](https://github.com/ros/rosdistro/pull/36216) to add libsimde-dev into rosdistro.

### CODE REVIEW

**Comment by hrp2-14** [comment]
Hi ! This project doesn't usually accept pull requests on the main branch.
If this wasn't intentionnal, you can change the base branch of this PR to devel
(No need to close it for that). Best, a bot.

**Comment by wep21** [comment]
@jcarpent @wxmerkt Are you fine with replacing `noetic` with `humble` in ros ci because `simde` is available since Ubuntu 22.04?

**Comment by wxmerkt** [comment]
If activated by default, it would be best to find a way to make sure it also works on Noetic since it's the most prevalent distribution right now (e.g. by vendoring). I don't think we should activate it by default if it cannot be built on the most common distribution

**Comment by wxmerkt** [comment]
Having thought about this some more, @wep21 we should be able to make the dependency conditional on the distro being rolling or humble - this way it can be activate for rolling/humble and inactive for older distros

File: package.xml [hunk 1]
```diff
@@ -13,8 +13,7 @@
   <build_depend>git</build_depend>
   <build_depend>doxygen</build_depend>
   <depend>eigen</depend>
-  <!-- There is no key for simde yet - deactivating for now. -->
-  <!-- <depend>simde</depend> -->
+  <depend>simde</depend>
```
[the remark this answers is not in the document]
Comment (wep21) [reply package.xml:16]: @wxmerkt Is the change as you suggested?

File: package.xml [hunk 2]
```diff
@@ -13,8 +13,8 @@
   <build_depend>git</build_depend>
   <build_depend>doxygen</build_depend>
   <depend>eigen</depend>
-  <!-- There is no key for simde yet - deactivating for now. -->
-  <!-- <depend>simde</depend> -->
+  <depend condition="$ROS_DISTRO == humble">simde</depend>
+  <depend condition="$ROS_DISTRO == rolling">simde</depend>
```
Comment (wxmerkt) [inline package.xml:17]: Thank you for the updates. Is it possible to use `depend="$ROS_DISTRO =='humble' || $ROS_DISTRO == 'rolling'`? And to leave a comment saying simde is not available for Noetic/20.04 or only available for 22.04 and following?

Comment (wep21) [reply package.xml:17]: > Thank you for the updates. Is it possible to use depend="$ROS_DISTRO =='humble' || $ROS_DISTRO == 'rolling'?
Yes, but considering upcoming ros2 release, is it better to use `condition="$ROS_DISTRO != noetic or $ROS_DISTRO != foxy"`? Do you have any thoughts?

> And to leave a comment saying simde is not available for Noetic/20.04 or only available for 22.04 and following?
Sure!

Comment (wxmerkt) [reply package.xml:17]: I am not sure if someone may not use this on other (even EOL) distros - so explicitly enabling where available might be the better choice? I am open for either though

Comment (wep21) [reply package.xml:17]: Thank you for your opinion.  I choose `condition="$ROS_DISTRO == humble or $ROS_DISTRO == rolling"`.

Comment (wep21) [reply package.xml]: @wxmerkt Please check [the change](https://github.com/Simple-Robotics/proxsuite/pull/178/commits/ce123e14a1d2eac2dbaf2476df3cfee031a9f657).

**Comment by wep21** [comment]
@wxmerkt @jcarpent By the way, is it possible to create ros2 release repository for this library and release it as a apt package?

File: package.xml (hunk 2 above)
Comment (wxmerkt) [reply package.xml]: Perfect, thank you!

**Review by jcarpent — APPROVED** [review]

**Pull request merged**